packages/core/build/eslint-rules/reserved-property-names.js

```diff
@@ -0,0 +1,236 @@
+/**
+ * This rule catches property overrides that conflicts with a key in the HTMLElement prototype
+ * Credit to https://github.com/ionic-team/stencil-eslint/blob/main/src/rules/reserved-member-names.ts
+ */
+const rule = {
+  meta: {
+    docs: {
+      description: 'This rule catches property overrides that conflicts with a key in the HTMLElement prototype',
+      category: 'Possible Errors',
+    },
+    schema: [],
+    type: 'problem',
+  },
+  create: function (context) {
+    return {
+      Decorator: node => {
+        const decoratorName = node.expression.callee.name;
+
+        if (decoratorName === 'property' || decoratorName === 'state' || decoratorName === 'event') {
+          const propName = node.parent.key.name;
+          if (isReservedProperty(propName)) {
+            context.report({
+              node: node.parent.key,
+              message: `"@${decoratorName} ${propName}" conflicts with a key in the HTMLElement prototype. Please choose a different name.`,
```

Comment:
This lint rule will help prevent accidental overrides of existing native properties